contrib/cindercompatibleapi/converter/snapshot.go

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound service.
+*/
+
+package converter
+
+import (
+	"errors"
+
+	"github.com/opensds/opensds/pkg/model"
+)
+
+// *******************Create a snapshot*******************
+
+// CreateSnapshotReqSpec ...
+type CreateSnapshotReqSpec struct {
+	Snapshot CreateReqSnapshot `json:"snapshot"`
+}
+
+// CreateReqSnapshot ...
+type CreateReqSnapshot struct {
+	VolumeID    string            `json:"volume_id"`
+	Name        string            `json:"name"`
+	Description string            `json:"description,omitempty"`
+	Force       bool              `json:"force,omitempty"`
+	Metadata    map[string]string `json:"metadata,omitempty"`
+}
+
+// CreateSnapshotRespSpec ...
+type CreateSnapshotRespSpec struct {
+	Snapshot CreateRespSnapshot `json:"snapshot,omitempty"`
+}
+
+// CreateRespSnapshot ...
+type CreateRespSnapshot struct {
+	Status      string            `json:"status"`
+	Description string            `json:"description"`
+	CreatedAt   string            `json:"created_at"`
+	Name        string            `json:"name"`
+	UserID      string            `json:"user_id"`
+	VolumeID    string            `json:"volume_id"`
+	Metadata    map[string]string `json:"metadata,omitempty"`
+	ID          string            `json:"id"`
+	Size        int64             `json:"size"`
+	UpdatedAt   string            `json:"updated_at"`
+}
+
+// CreateSnapshotReq ...
+func CreateSnapshotReq(cinderReq *CreateSnapshotReqSpec) (*model.VolumeSnapshotSpec, error) {
```

Comment:
I don't think this method should return error, since other methods only return ***RespSpec. You can package the validate operation in a standalone method.